### DISCUSSION TITLE
FR: Allow attaching current url to the screenshot

### DISCUSSION THREAD

**Discussion by elad2109** [flameshot-org/flameshot#2484] [Ideas]
Sometime we take a screenshot of a website.
Is there a way for the tool to automatically check if the screenshot is taken while the browser is on the whole screen and if so save it's url as a clickable attachment?

A viewer of the screenshot can click that url and it opens.
(We have an internal screenshot tool that does that in the company I work for).

If not possible automatically, can we allow manual clickable insertion of a url?

**Comment by mmahmoudian** [comment]
How can a picture have clickable area?! As far as I know it is impossible unless there is something like PDF. Most probably the software you have in your company does not only show a picture.

Also what you are asking in my opinion is way out if the scope of a screenshot tool. If one wants, they can easily use simple bash scripting and use imagemagic and tesseract to extract the URL if and only if:
1. The screenshot area covers the address bar
2. The address is short enough so that all of it fits in the screenshot 

To sum up:
1. Technically impossible to have a png or jpeg file with a clickable area
2. This is a niche usecase
3. This is out of the scope of a screenshot tool
4. It has practical issues as the URL typically overflows the visible addressbar

**Comment by elad2109** [comment]
Thanks,

You're right in my company it is never saved as an image rather a hosted image with a metadata section.
Is there any way we can think as saving the url (even manually) in the png meta data?

**Reply by mmahmoudian** [reply]
Of course, you can add textual data to PNG: https://www.w3.org/TR/PNG/#11textinfo
These are the fields you can have:


Title | Short (one line) title or caption for image
-- | --
Author | Name of image's creator
Description | Description of image (possibly long)
Copyright | Copyright notice
Creation Time | Time of original image creation
Software | Software used to create the image
Disclaimer | Legal disclaimer
Warning | Warning of nature of content
Source | Device used to create the image
Comment | Miscellaneous comment


You can use ImageMagic to add/alter comment field:

```sh
mogrify -comment "test" /somedir/somephoto.png
```

You can then view it using:

```sh
identify -verbose  /somedir/somephoto.png | grep "Comment:"
```

**Comment by elad2109** [comment]
Right, but I was looking for an easier way:

1) Maybe a plug-in that allows inserting "url" and it's saved to the png metadata automatically. 
2) Even better: a plug in that detects automatically if there is a browser window spread on the current window. So it allows optting in to add this data automatically to the saved png.

**Reply by mmahmoudian** [reply]
No plugin can do what you want simply because it should be able to get the URL in the active tab for you.

But this sounds like a quick weekend project to write something in bash for yourself. If you are on Linux and using X11, you can do the following (I assume you open your web browser in your first monitor):
1. take a screenshot and save it to a file with a specific name, (e.g `flameshot screen --number 1 --path ~/Pictures/screenshot.png`)
2. use `xdotool` to navigate to your browser's address bar by pressing <kbd>Ctrl</kbd>+<kbd>L</kbd>
3. wait for half a second to make sure the cursor is there
4. use `xdotool` to  invoke <kbd>Ctrl</kbd>+<kbd>A</kbd> to select the URL
5. wait
6. use `xdotool` to  invoke <kbd>Ctrl</kbd>+<kbd>C</kbd> to copy to clipboard
7. wait
8. use `mogrify -comment "$(xclip -out -selection clipboard)" ~/Pictures/screenshot.png` to add the URL to the comment section of the picture

hint: `xdotool key --delay=50 "ctrl+c"`

**Comment by elad2109** [comment]
I will go through your detailed reply.

I guess this could have been done if flamshot was a chrome extension. That's the gap.
OK
